client/state/memberships/coupon-list/test/actions.js

```diff
@@ -34,4 +61,193 @@ describe( 'actions', () => {
 			} );
 		} );
 	} );
+
+	describe( 'requestAddCoupon()', () => {
+		test( 'should dispatch add coupon failure and error notice on failure', async () => {
+			const siteId = 1;
+			const coupon = mockCoupon;
+			nock( 'https://public-api.wordpress.com' )
+				.post( '/wpcom/v2/sites/1/memberships/coupons' )
+				.replyWithError( { code: 'validation_error', message: 'Coupon code is already used' } );
+			const noticeText = 'Added coupon';
+			const dispatchedActions = [];
+
+			const dispatch = ( obj ) => dispatchedActions.push( obj );
+			await requestAddCoupon( siteId, coupon, noticeText )( dispatch );
+
+			const [ addAction, addFailureAction, noticeCreateAction ] = dispatchedActions;
+			expect( addAction ).toHaveProperty( 'coupon' );
+			expect( addAction ).toHaveProperty( 'siteId' );
+			expect( addAction ).toHaveProperty( 'type' );
```

Comment:
Instead of checking each property separately, I'd recommend something like:

```
expect( addAction ).toBe(
    expect.objectContaining( {
      coupon: expect.any( String ),
      siteId: expect.any( Number ),
      type: 'something',
    } ),
) 
```

This also would help resolve the issue @ciampo mentions a few lines below where we don't test for the property values.